crates/transaction-pool/src/validate.rs

```diff
@@ -80,10 +81,155 @@ pub trait TransactionValidator: Send + Sync {
+pub(crate) struct EthTransactionValidatorConfig<Client: AccountProvider> {
+    /// Chain id
+    chain_id: u64,
+    /// This type fetches account info from the db
+    client: Client,
+    /// Fork indicator whether we are in the Shanghai stage.
+    shanghai: bool,
+    /// Fork indicator whether we are using EIP-2718 type transactions.
+    eip2718: bool,
+    /// Fork indicator whether we are using EIP-1559 type transactions.
+    eip1559: bool,
+    /// The current max gas limit
+    current_max_gas_limit: u64,
+    /// gasprice
+    gas_price: Option<u128>,
+}
+
+#[async_trait::async_trait]
+impl<T: PoolTransaction + AccountProvider> TransactionValidator
+    for EthTransactionValidatorConfig<T>
+{
+    type Transaction = T;
+
+    async fn validate_transaction(
+        &self,
+        origin: TransactionOrigin,
+        transaction: Self::Transaction,
+    ) -> Result<TransactionValidationOutcome<Self::Transaction>, Error> {
+        let transaction = Arc::new(transaction);
+
+        // Checks for tx_type
+        match transaction.tx_type() {
+            LEGACY_TX_TYPE_ID => {
+                // Accept legacy transactions
+            }
+
+            EIP2930_TX_TYPE_ID => {
+                // Accept only legacy transactions until EIP-2718/2930 activates
+                if !self.eip2718 {
+                    return Err(Error::TransactionEip2930Disabled)
+                }
+            }
+
+            EIP1559_TX_TYPE_ID => {
+                // Reject dynamic fee transactions until EIP-1559 activates.
+                if !self.eip1559 {
+                    return Err(Error::TransactionEip1559Disabled)
+                }
+            }
+
+            _ => return Err(Error::TxTypeNotSupported),
+        };
+
+        // Reject transactions over defined size to prevent DOS attacks
+        if transaction.size() > TX_MAX_SIZE {
+            return Ok(TransactionValidationOutcome::Invalid(
+                transaction.clone(),
+                PoolError::OversizedData(*transaction.hash(), transaction.size(), TX_MAX_SIZE),
+            ))
+        }
+
+        // Check whether the init code size has been exceeded.
+        if self.shanghai {
+            match self.ensure_max_init_code_size(transaction.clone(), MAX_INIT_CODE_SIZE) {
+                Ok(_) => {}
+                Err(e) => return Ok(TransactionValidationOutcome::Invalid(transaction, e)),
+            }
+        }
+
+        // Checks for gas limit
+        if transaction.gas_limit() > self.current_max_gas_limit {
+            return Ok(TransactionValidationOutcome::Invalid(
+                transaction.clone(),
+                PoolError::TxExceedsGasLimit(
+                    *transaction.hash(),
+                    transaction.gas_limit(),
+                    self.current_max_gas_limit,
+                ),
+            ))
+        }
+
+        // Sanity check for extremely large numbers
+        if transaction.max_fee_per_gas() > Some((2 ^ 256) - 1) {
```

Comment:
use bit len instead.

Reply:
this is also not resolved.

but I realized, this check is actually redundant because we already enforce u128 and not u256, so we can actually remove those two checks.